addon/utils/selection.js

```diff
@@ -122,9 +128,11 @@ export default {
    * @param {Object[]} selectedItems - currently selected items
    * @param {Object} item - selection event target
    * @param {Object} rangeState - tracking the anchor and endpoint
+   * @param {Function} itemComparator - comparator for items
    */
-  specific (selectedItems, item, rangeState) {
-    const isCurrentlySelected = selectedItems.indexOf(item) >= 0
+  specific (selectedItems, item, rangeState, itemComparator) {
+    const index = selectedItems.findIndex(selectedItem => itemComparator(selectedItem, item))
+    const isCurrentlySelected = (index >= 0)
```

Comment:
Can't use .removeObject in most places where you use a comparator as the objects are not guaranteed to be the same even if they have the same contents.